x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -23,15 +24,17 @@ import {
   sendGetAgentConfigList,
 } from '../../policy/store/policy_list/services/ingest';
 import { AGENT_CONFIG_SAVED_OBJECT_TYPE } from '../../../../../../ingest_manager/common';
+import { startPoll, POLL_INTERVAL } from '../../../common/polling';
 
 export const endpointMiddlewareFactory: ImmutableMiddlewareFactory<EndpointState> = (coreStart) => {
+  // eslint-disable-next-line complexity
   return ({ getState, dispatch }) => (next) => async (action) => {
     next(action);
     const state = getState();
 
     // Endpoint list
     if (
-      action.type === 'userChangedUrl' &&
+      (action.type === 'userChangedUrl' || action.type === 'appRequestedEndpointList') &&
```

Comment:
I'm keeping track of when polling was started or stopped with the `isAutoRefreshEnabled` state and this should stop multiple polls from occurring.

It's true that `userChangedUrl` and `appRequestedEndpointList` are disjoint actions.  I see what you're saying where I could split these out as 2 separate cases to avoid multiple polls.  However, the url can update multiple times on this page just by changing the page index, etc, so I'd still need to make sure I'm not triggering multiple polls.

Reply:
That is also true, the only thing I'm indicating that triggering 'appRequestedEndpointList' action can also invoke something that is related to 'userChangedUrl' and maybe is not desirable to be invoked at that point. But in any case as mentioned I'm still not having the complete picture of the flow so might be overthinking it, feel free to ignore this comment.

Reply:
🤔 filtering could in deed introduce some weird UI behaviors while both calls are in flight. I think when we get there it may be mitigated by assigning a "transaction id" everytime a API call is made and then only update state if the transaction id is equal to the latest value.

Actually: do we already have that (edge) case now? when user changes items per page or paginates while a refresh is on going? or when a "auto-refresh" request is issued right after one of these pagination/perPage items are change?

Reply:
@paul-tavares yea, I think we may need to revisit this, I think this middleware file needs a look regardless, especially when it comes to `useChangedUrl`.

The transaction id sounds like a good idea.  I just wonder which behavior we should expect when the middleware makes decisions based on `selectors`.  Do we actually want it to operate on potentially stale state?

Reply:
yeah - let's open some tech. debt issue for follow up. this is ok for now, I think 👍